viz_scripts/scaffolding.py

```diff
@@ -137,14 +148,93 @@ def energy_intensity(df,df1,distance,col1,col2):
     return df
 
 
+def cost(data, cost, dist, repm, mode):
+    """
+    Calculates the cost of the CanBikeCO E-bike pilot program
+    
+    Parameters:
+        data - CanBikeCO data input
+        cost - dataframe defining cost ($/PMT) for each mode
+        dist - feature name in data of feature with distance in miles
+        repm - feature name in data of feature with replaced mode
+        mode - feature name in data of feature with confirmed mode
+        
+    Returns:
+        data with appended cost feature for each trip in $$$ for both mode and replaced mode (float)
+    """
+
+    # Create a copy of the cost dataframe
+    cost = cost.copy()
+
+    # Create a replaced mode column in cost same as mode
+    cost[repm] = cost['mode']
+
+    # Pair cost with mode
+    dic_cost__trip = dict(zip(cost[repm],cost['C($/PMT)']))
+    
+    # Create new features in data for replaced mode
+    data['cost__trip_'+repm] = data[repm].map(dic_cost__trip)
+    
+    # Create new features in data for confirmed mode
+    cost[mode] = cost[repm]
+    dic_cost__trip = dict(zip(cost[mode],cost['C($/PMT)']))
+    data['cost__trip_'+mode] = data[mode].map(dic_cost__trip)
+    
+    return data
+
+
+def time(data, dura, dist, repm, mode):
+    """
+    Calculates the cost of the CanBikeCO E-bike pilot program
+    
+    Parameters:
+        data - CanBikeCO data input
+        dura - dataframe defining duration ((1/speed)/PMT) for each mode
+        dist - feature name in data of feature with distance in miles
+        repm - feature name in data of feature with replaced mode
+        mode - feature name in data of feature with confirmed mode
+        
+    Returns:
+        data with appended cost feature for each trip in $$$ for both mode and replaced mode (float)
+    """
+
+    # Create a copy of the dura dataframe
+    dura = dura.copy()
+
+    # Create a replaced mode column in dura same as mode
+    dura[repm] = dura['mode']
+
+    # Pair dura with mode
+    dic_dura__trip = dict(zip(dura[repm],dura['C($/PMT)']))
```

Comment:
not sure why you are using `$/PMT` here...

Reply:
Right, that needed to be fixed.